### DISCUSSION TITLE
Is there a way to have an external link with `RouteLocationNormalized` object syntax

### DISCUSSION THREAD

**Discussion by Barbapapazes** [vuejs/router#2125] [Help and Questions] [answered]
Hello,

see https://github.com/nuxt/nuxt.com/pull/1487
see https://github.com/nuxt/nuxt/issues/25207

I'm looking for a way to manage query using the object syntax of `RouterLink` for an external link.

If you do `to="https://example.com?query=123"`, it's ok but I prefer the object syntax for a better readability. I would prefer

```
to={
	path: "https://example.com",
	query: {
		query: 123
	}
}
``` 

This give me `/https://...`. As you can see, there is a first `/`. I think it's a wanted behavior since the name of the parameter is `path` but couldn't be an expected behavior to have a built-in external link management?

**Comment by posva** [comment, marked as answer]
Handling external links is intentionally out of scope of the router. You could always build these utilities yourself with `new URL()` and `new URLSearchParams()`. These APIs should make it fairly easy to build
